src/pika_server.cc

```diff
@@ -1206,7 +1208,82 @@ void PikaServer::AutoCompactRange() {
   }
 }
 
-void PikaServer::AutoPurge() { DoSameThingEveryDB(TaskType::kPurgeLog); }
+void PikaServer::AutoBinlogPurge() { DoSameThingEveryDB(TaskType::kPurgeLog); }
+
+void PikaServer::AutoServerlogPurge(){
+std::string log_path = g_pika_conf->log_path();
+  int retention_time = g_pika_conf->log_retention_time();
+  if (retention_time <= 0) {
+    return;
+  }
+  std::vector<std::string> log_files;
+
+  if (!pstd::FileExists(log_path)) {
+    return;
+  }
+
+  if (pstd::GetChildren(log_path, log_files) != 0) {
+    return;
+  }
+
+  time_t t = time(nullptr);
+  struct tm* now = localtime(&t);
+  int now_year = now->tm_year + 1900;
+  int now_month = now->tm_mon + 1;
+  int now_day = now->tm_mday;
+
+  //logformat: pika.[hostname].[user name].log.[severity level].[date].[time].[pid]
+  for (const auto& file : log_files) {
+    if (file.substr(0, 5) != "pika.") {
+      continue;
+    }
+
+    size_t log_pos = file.find(".log.");
+    if (log_pos == std::string::npos) {
+      continue;
+    }
+
+    // Start at the end of ".log." to find the date and time
+    size_t date_pos = file.find('.', log_pos + 5);
+    if (date_pos == std::string::npos || date_pos + 16 > file.length()) {
+      continue;
+    }
+
+    std::string date = file.substr(date_pos + 1, 8);
+    std::string time = file.substr(date_pos + 10, 6);
+
+    int log_year = std::atoi(date.substr(0, 4).c_str());
+    int log_month = std::atoi(date.substr(4, 2).c_str());
+    int log_day = std::atoi(date.substr(6, 2).c_str());
+
+    struct tm log_time;
+    struct tm now_time;
+
+    log_time.tm_year = log_year - 1900;
+    log_time.tm_mon = log_month - 1;
+    log_time.tm_mday = log_day;
+    log_time.tm_hour = 0;
+    log_time.tm_min = 0;
+    log_time.tm_sec = 0;
+
+    now_time.tm_year = now_year - 1900;
+    now_time.tm_mon = now_month - 1;
+    now_time.tm_mday = now_day;
+    now_time.tm_hour = 0;
+    now_time.tm_min = 0;
+    now_time.tm_sec = 0;
+
+    int64_t log_timestamp = mktime(&log_time);
+    int64_t now_timestamp = mktime(&now_time);
+    int64_t interval_days = (now_timestamp - log_timestamp) / 86400;
+
+    if (interval_days > retention_time) {
```

Comment:
A logic is needed to guarantee that the latest log(which is in use) will not be deleted even if it's  logically expired. 
The time extracted from log file name is the time of it's creation. thus it's possible that a log has logically expired but still in use if the pika server is inactive.

Reply:
Do your mean that assuming a Pika instance has been running continuously for 8 days, a server log created 8 days ago should logically have expired, but in reality, it is the most recent and currently used server log?

Reply:
yes, that's what i mean